test/e2e/multicast_test.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"antrea.io/antrea/pkg/agent/config"
+	"antrea.io/antrea/pkg/features"
+)
+
+func skipIfMulticastDisabled(tb testing.TB) {
+	skipIfFeatureDisabled(tb, features.Multicast, true, false)
+}
+
+func TestMulticast(t *testing.T) {
+	skipIfHasWindowsNodes(t)
+	skipIfNotIPv4Cluster(t)
+	skipIfMulticastDisabled(t)
+
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+
+	t.Run("testMulticastBetweenPods", func(t *testing.T) {
+		skipIfEncapModeIsNot(t, data, config.TrafficEncapModeNoEncap)
+		testMulticastBetweenPods(t, data)
+	})
+}
+
+func testMulticastBetweenPods(t *testing.T, data *TestData) {
+	senderName, _, cleanupFunc := createAndWaitForPod(t, data, data.createMcJoinPodOnNode, "test-server-", "", testNamespace, false)
```

Comment:
I think the test-server should be the one joining a Multicast group, and waiting for the Multicast traffic, so is it better to name it as "test-client-" ?

Reply:
fixed